### PR TITLE
Add DEHB

### PR DESCRIPTION
**Pull request by HideakiImamura**
## Contributor Agreements

Please read the [contributor agreements](https://github.com/optuna/optunahub-registry/blob/main/CONTRIBUTING.md#contributor-agreements) and if you agree, please click the checkbox below.

- \[x\] I agree to the contributor agreements.

## Motivation

Differential Evolution with Hyperband (DEHB) is a recently introduced algorithm which uses differential evolution and hyperband for sampling and pruning. It is easy to implement and computationally cheap, so I will introduce it in the OptunaHub. And, I expect that it is superior to BOHB (TPE + Hyperband) of Optuna.

## Description of the changes
- Add `DEHBSampler` and `DEHBPruner`

### CODE REVIEW

**Comment by HideakiImamura** [comment]
The CI fails due to irrelevant changes in this PR.

**Comment by HideakiImamura** [comment]
@y0z Could you review this PR?

File: package/samplers/dehb/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+author: Hideaki Imamura
+title: Differential Evolution with Hyperband (DEHB) Sampler
+description: A sampler for differential evolution with Hyperband.
+tags: [sampler, pruner, hyper parameter optimization, differential evolution, hyperband]
```
Comment (y0z) [inline package/samplers/dehb/README.md:5]:
```suggestion
tags: [sampler, pruner, hyperparameter optimization, differential evolution, hyperband]
```

Another option is "hyper-parameter optimization."

File: package/samplers/dehb/sampler.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+        ret = self._random_sampler.sample_independent(study, trial, param_name, param_distribution)
+        return ret
+
+    def infer_relative_search_space(
+        self, study: Study, trial: FrozenTrial
+    ) -> dict[str, BaseDistribution]:
+        search_space: dict[str, BaseDistribution] = {}
+        for name, distribution in self._search_space.calculate(study).items():
+            if distribution.single():
+                # `cma` cannot handle distributions that contain just a single value, so we skip
```
Comment (y0z) [inline package/samplers/dehb/sampler.py:116]: Perhaps should this line be as follows?
```suggestion
                # `DEHBSampler` cannot handle distributions that contain just a single value, so we skip
```

**Review by y0z — COMMENTED** [review]
I leave some comments.

I also confirmed that this sampler works in my envorinment.
slice-dehbsampler
step-dehbsampler

history-dehbsampler

File: package/samplers/dehb/sampler.py (hunk 2 above)
Comment (HideakiImamura) [reply package/samplers/dehb/sampler.py:116]: I just removed the comments. This comments come from the implementation of Optuna's CmaEsSampler.

**Comment by HideakiImamura** [comment]
@y0z Thanks for your reviews. I applied all of your suggestions. PTAL.

**Review by y0z — APPROVED** [review]
LGTM

**Comment by HideakiImamura** [comment]
Let me merge this PR since we have one approval.

**Pull request merged**